src/Mono.Android/Android.Runtime/JNIEnv.cs

```diff
@@ -152,15 +152,16 @@ internal static unsafe void Initialize (JnienvInitializeArgs* args)
 		{
 			Logger.Categories = (LogCategories) args->logCategories;
 
-			var __start = new DateTime ();
+			Stopwatch stopper = null;
+			long elapsed, totalElapsed = 0;
 			if (Logger.LogTiming) {
-				__start = DateTime.UtcNow;
-				Logger.Log (LogLevel.Info,
-						"monodroid-timing",
-						"JNIEnv.Initialize start: " + (__start - new DateTime (1970, 1, 1)).TotalMilliseconds);
-				Logger.Log (LogLevel.Info,
-						"monodroid-timing",
-						"JNIEnv.Initialize: Logger JIT/etc. time: " + (DateTime.UtcNow - new DateTime (1970, 1, 1)).TotalMilliseconds + " [elapsed: " + (DateTime.UtcNow - __start).TotalMilliseconds + " ms]");
+				stopper = new Stopwatch ();
```

Comment:
Is using `Stopwatch` better than the `DateTime` previously used?  `Stopwatch` is a heap-allocated type vs. a struct, so I feel like this will increase memory use in "timing-enabled" apps, without necessarily improving time resolution.

Reply:
It has much less overhead, though and I think less impact on performance is important here.